# running log of reviews of Brucella readings

## Brucella in Australia
[Cook et. al, 1966](http://onlinelibrary.wiley.com/doi/10.1111/j.1751-0813.1966.tb04603.x/abstract)  
Cook, I., R. W. Campbell, and G. Barrow. 1966. Brucellosis in North Queensland rodents. Australian veterinary journal 42:5–8.  

This paper introduced the identification of *Brucella* in rodents trapped in Jordan Creek near Palmerston National Park, located in along the east coast of central Queensland, Australia. There was little domestic cattle used in the area, though wild pigs were known to inhabit the grounds. Spleen and liver tissues were extracted from trapped rodents - 6 different species in all - and used as source material to innoculate mice with; of the 54 total rodents, 30 were *Rattus assimilis* and 16 were *Meloymys cervinipes*. A total of 7 *Brucella* strains were derived from three species: *Rattus assimilis* (4), *Melomys cervinipes* (2), and *M. lutilllus* (1).  

- Serum from putatively infected mice provided serum with which agglutination tests for a variety of microorganisms were conducted, including *B. abortus*, *Coxiella burneii*, *Proteus* OX19, and several strains of *Leptospira* and *Salmonella*. Agglutination tests were *positive for *B. militensis**.  
- Further microbiological tests includded cross immunity for *Rikettsia tsutsugamushi*, tail blood smear observations for *Eperythrozoon cocoides* and *Haemobartonella muris*, and heart blood smear for *Spirillum minus*. Inflammed spleens from infected mice were used as sources to isolate potential *Brucella* pathogens. While *E. coccoides* was identified in two of these mice, eight total isolations were identified. Seven of eight of these isolates grew on Albimi agar plates, in both 10% CO<sub>2</sub> and air (both at 37 °C). Oxidative metabolic tests indicated utilization of all eight amino acids and three carbohydrate sources tested, indicating that they are **most likely classified as *B. suis* type 3** strains.  
- Guinea pigs were infected with four of the isolates; all died between 12-28 days. Serum agglutinated to *B. abortus* antigen.

This paper proved significant because it speculated that rodents may serve as a natural reservoir with which domestic livestock may become infected. Pigs were known to eat rodents; the question is whether or not this consumption would prove to transmit one *Brucella* strain from one host to a different (presumably mammalian) host. What was unclear was whether the rodents were reservoirs of the pathogen, or if they became infected by the pigs initially. A few other natural history notes: 
- *B. suis* is transmitted in pigs when swine suffer abortions
- *B. suis* is also transmitted frequently through urine, making it likely that a herd is heavily contaminated in domestic conditions

## Brucella in Rodents  

### [Meyer, 1976](http://www.sciencedirect.com/science/article/pii/0093691X76900194)  

Meyer, M. E. 1976. Evolution and taxonomy in the genus Brucella : Brucellosis of rodents. Theriogenology 6:263–272.  
This was a follow up to Cook et al., 1966, which had left unclear whether Brucella observed in rodents served as the reservoir species with which domesticated animals acquired the pathogen, or if pigs, cows, etc. were tranmitting the pathogen to the rodents. The scientist examined 19 extant strains from North America, Africa, and Australia isolated over the past 20 years. Additional data from USSR was incorporated into the analsyis from prior reports.  
*Brucella* were compared using classic (deterministic) microbiological tests as well as oxidative metabolic patterns of 39 amino acids and carbohydrates. Observations are summarized as follows:
- Metabolic patterns suggested strains from rodents from Africa and Austraila **are similar to *B. suis* ** because that strain is the only one in the genus which can oxidize substances of the urea cycle. *B. susis* strains reported from the USSR also demonstrated such metabolic activity. However, the *B. neotomae* strains do not metabolize these amino acids. **This suggests that all rodent-obtained *Brucella* with the exception of *B. neotomae* are similar to *B. suis* **.
- Determinative methods similarly suggested that **all *Brucella* obtained from rodents are similar to *B. susis* WITH THE EXCEPTION of *B. neotomae* **. This distinction among isolates had to do with a pair of tests, though I didn't find these as conclusive as the binary metabolic results noted above:
  - the concentration of phage which a strain woudd lyse
  - the agglutinability in monospecific antisera
  - hydrogen sulfide production  
** Each of these patterns can vary among *B. suis* strains, which makes me think that a genomic approach may reveal different patterns of shared ancestry**.  

Meyer presents a few interesting pieces of information to contrast:  
- Some *Brucella* are extremely similar with respect to DNA:DNA hybridization studies (94-100%), while others are not particularly similar for species within the same genera (only ~70% similar)
- Phenotypic plasticity persists despite such DNA similarity. There is diversity in host range of infectivity, physilogical and biochemical characteristics, and morphology.

Meyer, on pages 265-266, suggests that researchers have split into two camps, of which many of these questions could be tested...    
1) *B. suis* and *B. neotomae* share a common ancestor, and it was a rodent:  
- That ** *B. neotomae* represents a sister clade to *B. suis* which shared an ancestral strain circulated in rodents before hoofed animals diverged from rodents**. From what I can gather, that's an indication that *Brucella* have been with Mammals a really long time
  - this split is around 100 million years ago between the hooved decendents (Laurasiatheria) and the rodents (Rodentia). Check out [this paper](http://www.utheria.org/uploads/nature05634.pdf) for details. 
-  ** *B. suis* is really a broader collection of both *B. suis* and *B. neotomae* - they shared a common ancestor while *B. neotomae* represents tha archetypal rodent Brucella** - it was *B. suis* before *B. suis* was transmitted into hooved animals. 
  - One thing you can test, then, is gene loss. One would expect that *B. suis* strains share more functional genes with *B. neotomae* than other strains, **but what about B. canis**?
- ** *B. neotomae* is from America** because it was obtained from a US sample. WHAT?!? We could totally do a bit of digging on this one...
- *B. neotomae* doesn't appear to do well in other hosts because it appears "rough"... What does morphological shape have anything to do with transmissibility? Or host infectivity?  

2) *B. suis* are the ancestral state, and any rodent-associated Brucella strain is derived from a domesticated hooved animal
- Rodents life cycle are too short to circulate Brucella in the wild. Because you only find Brucella in rodents in livestock and not otherwise, this suggests that Rodents are the sink, not the source, of the pathogen.
- Rodents require doses of infection too large to accumulate in a large number of animals to persist just within the population itself. Rodents couldn't give this to other rodents. ... *what about in milk though?*  

**The final takeaway** - Meyer seems to side with Rodent-associated Brucella as being derived from another source. It's not ancestral, except for *B. neotomae*, which they don't have a specific answer for its origin, other than it is the only example of a rodent-speciic organism.  


### [Tiller, 2010](http://aem.asm.org/content/76/17/5837.full)  

Tiller, R. V., J. E. Gee, M. A. Frace, T. K. Taylor, J. C. Setubal, A. R. Hoffmaster, and B. K. De. 2010. Characterization of Novel Brucella Strains Originating from Wild Native Rodent Species in North Queensland, Australia. Applied and Environmental Microbiology 76:5837–5845.  

As a follow up to the preliminary work by Cook et. al, this work investigated the seven Brucella isolates by both recapitulating the old approaches taken in the 1960's combined with a suite of relatively modern genomic approaches; their methods included:  
- **Phenotypic characterization:** All seven strains grew on selective media under similar conditions reported earlier by Cook. Biochemical reactions initially reported by Cook and Meyer were similarly confirmed by Tiller. Biotyping *Brucella* spp. using similar historic diagnostics including:
  - susceptibility to Tbilisi phage
  - agglutination profiles
  - thionine and fushin dye growth  

 In short, these measures repeated earlier findings that characterized rodent *Brucella* as phenotypically unique from other non-rodent isolates.

- **PCR profiles**: All seven strains hybridized to a probe developed for the 4-bp substitution in the 16-S gene sequence unique to the BO1 and BO2 strains. In addition, the BRU probe used to detect classic *Brucella* strains did not hybridize. These results suggest that the Australian rodent samples in question are more similar to the BO1 and BO2 strains than classic *Brucella* species.  

- **gene sequencing:** A suite of genes were sequenced with Sanger technique. These were reported in two parts:
  - **16S rRNA, *recA*, and *rpoB* genes**  
  Full lenth sequences of each gene was sequenced for each of the seven rodent isolates; a consensus sequence was established for each gene among the isolate sequences, and compared to consensus sequences for *Brucella* spp. as well as BO1 and BO2 isolates. They find:
    - 16S rRNA sequence among rodent consensus match BO1 and BO2, and differ at an additional nucleotide position (position 308) among all non-rodent strains (BO1, BO2, and *Brucella* spp.)
    - *recA* consensus among rodent strain demonstrated 99.2% identity to *Brucella* spp., **yet just 97.9% and 98.7% identity to BO1 and BO2 respectively**.
    - *rpoB* suggested 100% identity to the rodent-isolate strain 83-13, and evidently supports a closer identity of rodent strain consensus to BO1 than BO2.  

   These results indicate a mixed interpretation: single-gene samples produce competing ancestral hypotheses (ie. different trees).   
   
  - **MLSA via houskeeping genes:** To circumvent the mixed results of the individual gene trees above, a single tree was built using a series of housekeeping genes. As before, individual rodent isolates were sequenced and a consensus sequence was constructed. Using this approach (with 9 other genes) the consensus Australian rodent sequence shared 100% identity to *Brucella* sp. strain 83-13. The next highly related strain was BO2 (98.7%).  

- **MLVA:** Each Australian rodent isolate had a unique profile of variable number tandem repeats. However, the individual isolates form a distinct clade separate from the classic *Brucella*. spp.  

- **MUMi distances:** They used 454 pyrosequencing on one of the seven Australian rodent isolates - **strain NF 2653** - and looked at the core genomic DNA regions to assess inra and interspecies variability. MUMi provides a series of pairwise comparisons to evaluate the likelihood of identity between two genomes. Comparisons to the *B. inoptina* BO1 and BO2 strains, as well as the rodent-associated strain *Brucella* sp. 83-13, as well as *B. neotomae* 5K33, *B. microti*, and *B. suis* 686 (WHO *B. suis* biovar 3) were included for comparison - this select group was identified because each has been isolated from a rodent. They conclude that BO1 and BO2 should be grouped together, **while the *Brucella* sp. 83-13 and the representitive Australian sample should be groupd together**.

**Takeaways**
- I'm unclear why they incorporate *Brucella* sp. 83-13 in some analyses but not in others. When they talk about the MLVA and MUMi work it's included, but when they do the MLSA and single-gene tree building it's exluded - what gives?  
- Mixed results occur when examining phylogenetic relationships using single-gene trees. Even their MLSA approach using multiple genes is suspicious - while they get 100% identity to a rodent-isolate strain (*Brucella* sp. 83-13), they used a consensus of the Australian rodent isolates to conduct these tests. I'm unclear why they didn't just build trees from individual strains and identify where along the tree these seven individual isolates clustered.
- The consensus Australian rodent sequence shows significant sequence divergence from all other B. suis strains used in their large Brucella family tree. **I'm pretty sure we get mixed results depending on which strain is input in our data**.  


incorporating a handful of gene sequences and multi-locus sequence typing (MLST). They indicate that the sequences are distinct from *Brucella spp.* and likely represent a unique Brucella species; in addition, while the gene sequences of the seven isolates were highly similar, MLST results indicated enough variation to support a monophyletic clade to group each of these isolates. 

Side note: the seven Brucella isolates are now part of a broader CDC collection in Atlanta
