# installs
## core stuff
```
sudo apt-get update
sudo apt-get -y upgrade
sudo apt-get -y install tmux git curl gcc make g++ unzip default-jre libdatetime-perl libxml-simple-perl libdigest-md5-perl bioperl
```

## roary (pangenome stuff)
```
sudo apt-get -y install bedtools cd-hit ncbi-blast+ mcl parallel cpanminus prank mafft fasttree
sudo cpanm -f Bio::Roary
```

## prokka dependencies
### hmmer-3
```
mkdir hmmer
cd hmmer/
wget http://eddylab.org/software/hmmer3/3.1b2/hmmer-3.1b2-linux-intel-x86_64.tar.gz
tar xvzf hmmer-3.1b2-linux-intel-x86_64.tar.gz
cd hmmer-3.1b2-linux-intel-x86_64
sudo ./configure
make
sudo make install
```
To be safe just put a link to these binaries in your `bin`:
```
cd /bin
sudo ln -s /home/ubuntu/hmmer/hmmer-3.1b2-linux-intel-x86_64/binaries/* .
```

### signalP
Go to the website's [download page](http://www.cbs.dtu.dk/cgi-bin/nph-sw_request?signalp) and enter all your information. Enter a university (.edu) email address. Click the link from the email generated by your request and copy the .tar.gz file to your server. From there, follow their instructions outlined in the `signalp-4.1.readme` file once you've decompressed the file archive. For simplicity, I've put a copy of that file on my Github page so that I can use that as needed:
```
cd $HOME
wget https://github.com/devonorourke/brucella/raw/master/signalp-4.1f.Linux.tar.gz
tar xvzf signalp-4.1f.Linux.tar.gz
```
Then open up that directory and edit the `signalp` file with this info:
```
# FULL PATH FOR signalp-4.1 directory
/home/ubuntu/signalp-4.1
```
Place a link to the program file `signalp` in the `/bin`:
```
cd /bin
sudo ln -s /home/ubuntu/signalp-4.1/singalp .
```

### rRNA predictor 'barrnap'
```
cd $HOME 
wget https://github.com/tseemann/barrnap/archive/0.8.tar.gz
tar -xzvf 0.8.tar.gz
cd /bin
sudo ln -s /home/ubuntu/barrnap-0.8/bin/barrnap
```

## prokka itself
Download to home directory and place a link to binary in `/bin`
```
cd $HOME
git clone https://github.com/tseemann/prokka.git
cd prokka/bin/
./prokka --setupdb
sudo ln -s /home/ubuntu/prokka/bin/prokka /bin/.
```

# lets run some programs!

## actually run prokka
I transferred all .fasta files from the **broad** external references as well as all the Australian samples. These were put in a single `fastas` directory within the `$HOME` directory. All files were then annotated with Prokka as noted in the shell script **prokka.sh** below. Make sure to first make the shell script executable before running (`chmod +x prokka.sh`)!

```
#!/bin/bash
cd /home/ubuntu/fastas
SAMPLE_LIST=`ls *.fasta | sed 's/.fasta//' | sort -u`
for SAMPLE in $SAMPLE_LIST
do
	prokka \
	--outdir ${SAMPLE} \
	--force \
	--prefix ${SAMPLE} \
	--cpus 0 \
	${SAMPLE}.fasta
done
```


## actually run Roary
See more details about [install](https://github.com/sanger-pathogens/Roary/blob/master/README.md) and [executing](https://github.com/microgenomics/tutorials/blob/master/pangenome.md) scripts.  
First thing to do is create sym links for the .gff files from the Prokka output into a new `Roary` directory. Once those are in place you can execute their pretty generic command:
```
roary -e -n -v *.gff
```


